Sprint 15

ZeroXKiritsu opened the pull request.

Hardwired8403 committed:
feat: Add pagination to findAll and findAllOwner methods in BookingService

Adding pagination to the `findAll` and `findAllOwner` methods in the `BookingService` interface and implementing it in the `BookingServiceImpl` class. The `findAll` method now takes additional parameters `from` and `size` for specifying the starting index and size of the result list. The same changes have been made to the `findAllOwner` method.

sergeymamonov commented on src/main/java/ru/practicum/shareit/booking/dto/BookingDto.java:
```diff
@@ -12,7 +13,8 @@
 @NoArgsConstructor
-public final class BookingDto {
+@Builder
+public class BookingDto {
```
Dto это неизменяемый класс, поэтому можно добавить модификатор final к определению класса и полям. А также лучше отказаться от аннотации @Data, так как в ней собрано несколько аннотаций, включая @Setter, то есть модифицирующий метод.

sergeymamonov commented on src/main/java/ru/practicum/shareit/booking/service/BookingServiceImpl.java:
```diff
@@ -180,8 +185,7 @@ private Booking validateBookingDetails(Long userId, Long bookingId, Integer numb
                     throw new NotFoundException("Пользователь не владелeц и не автор бронирования ");
                 }
                 return booking;
-            default:
```
Почему удалена дефолтная ветка?

sergeymamonov commented on src/main/java/ru/practicum/shareit/request/ItemRequestMapper.java:
```diff
@@ -0,0 +1,46 @@
+
+@UtilityClass
+public class ItemRequestMapper {
+    public ItemRequest toRequest(User user, ItemRequestDto itemRequestDto) {
```
Используется ли где-то параметр user?

sergeymamonov commented on src/test/java/ru/practicum/shareit/request/ItemRequestControllerTest.java:
```diff
@@ -0,0 +1,122 @@
+
+    @Test
+    @SneakyThrows
+    void get() {
```
Не очень информативное название для теста.

sergeymamonov commented on src/test/java/ru/practicum/shareit/request/RequestServiceIT.java:
```diff
@@ -0,0 +1,57 @@
+@Transactional
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.NONE)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+public class RequestServiceIT {
```
А почему у остальных тестовых классов есть окончание Test. А у этих классов нет окончания Test?

Hardwired8403 committed:
Refactor BookingDto and ItemRequestMapper, Add ItemRequestControllerTest

The changes in this commit include:
- Refactor the `BookingDto` class by replacing the `@Data` annotation with `@Getter` and `@Setter`.
- Refactor the `ItemRequestMapper` class by removing the `User` parameter in the `toRequest` method.
- Update the `ItemRequestServiceImpl` class to use the modified `ItemRequestMapper.toRequest` method.
- Rename the `src/test/java/ru/practicum/shareit/request/RequestServiceIT.java` file to `src/test/java/ru/practicum/shareit/request/RequestServiceTest.java`.
- Add tests to the `ItemRequestControllerTest` class.

Refactor BookingDto, ItemRequestMapper, Add ItemRequestControllerTest

Hardwired8403 committed:
feat: Update BookingDto with lombok annotations

This update introduces lombok annotations in the `BookingDto` class, reducing boilerplate code. The `@Data` annotation is used to replace `@Getter`, `@Setter`, `@AllArgsConstructor`, `@NoArgsConstructor` and `@Builder` annotations.

Hardwired8403 committed:
Refactor BookingDto class structure

The file BookingDto.java in the src/main/java/ru/practicum/shareit/booking/dto directory has been modified. The changes include refactoring the class structure of BookingDto by adding annotations from the lombok library such as @Data, @NoArgsConstructor, @AllArgsConstructor, and @Builder. These annotations help in generating boilerplate code for data classes automatically.

sergeymamonov commented on src/main/java/ru/practicum/shareit/booking/dto/BookingDto.java:
```diff
@@ -12,6 +13,7 @@
+@Builder
 public final class BookingDto {
     @NotNull
     private Long itemId;
```
Сюда тоже лучше добавить модификатор final.

sergeymamonov commented on src/main/java/ru/practicum/shareit/item/dto/ItemDto.java:
```diff
@@ -11,12 +12,19 @@
 @NoArgsConstructor
-public final class ItemDto {
+@Builder
+public class ItemDto {
```
final

sergeymamonov commented on src/main/java/ru/practicum/shareit/item/dto/ItemDto.java:
```diff
@@ -11,12 +12,19 @@
+@Builder
+public class ItemDto {
     @NotBlank
     private String name;
```
final. Эти модификаторы лучше добавить ко всем dto.

sergeymamonov commented on src/test/java/ru/practicum/shareit/item/ItemServiceIT.java:
```diff
@@ -0,0 +1,130 @@
+@Transactional
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.NONE)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+public class ItemServiceIT {
```
Аналогично. Непонятно, почему у этого класса нет окончания Test? И у остальных тестовых классов UserServiceIT и так далее.

Hardwired8403 committed:
Feat: Modify class names and update imports

This commit renames the classes `CommentDto`, `ItemDto`, `ItemRequestDto`, and `UserDto` to make them final. The `BookingServiceIT` and `ItemServiceIT` test classes have also been renamed to `BookingServiceTest` and `ItemServiceTest` respectively. These changes ensure that the classes are not extended or overridden, providing better encapsulation and immutability.
